src/main.js

```diff
@@ -383,60 +381,31 @@ function handleAppWebContentsCreated(dc, contents) {
 
   contents.on('will-navigate', (event, url) => {
     const contentID = event.sender.id;
-    const parsedUrl = new URL(url);
-    const urlIsTrusted = isTrustedURL(parsedUrl);
-    const urlIsTrustedExternalLoginPath = parsedUrl.hostname === customLogins[contentID].externalHostname;
+    const parsedURL = parseURL(url);
+    const urlIsTrusted = isTrustedURL(parsedURL);
 
     // don't prevent custom login attempts (oath, saml)
-    if (!urlIsTrusted && !urlIsTrustedExternalLoginPath) {
+    if (!urlIsTrusted && !customLogins[contentID].inProgress) {
       event.preventDefault();
-
-      // reset custom login if in progress
-      if (customLogins[contentID].inProgress) {
-        customLogins[contentID].inProgress = false;
-        customLogins[contentID].externalHostname = null;
-
-        // redirect to starting url if set
-        if (customLogins[contentID].startingURL) {
-          event.sender.loadURL(customLogins[contentID].startingURL);
-          customLogins[contentID].startingURL = null;
-        }
-      }
     }
   });
 
   // handle custom login requests (oath, saml):
-  // 1. are we navigating to a supported local custom login path from the `/login` page? (did-start-navigation listener)
-  //    - indicate custom login is in progress and store starting point for possible reset
-  // 2. is a custom login in progress but we don't have the 3rd party hostname yet? (will-redirect listener)
-  //    - store 3rd party hostname to trust subsequent navigation changes within that hostname
-  // 3. are we finished with the custom login process? (did-start-navigation listener)
-  //    - indicate custom login is NOT in progress and clear any stored 3rd party hostname's
+  // 1. are we navigating to a supported local custom login path from the `/login` page?
+  //    - indicate custom login is in progress
+  // 2. are we finished with the custom login process?
+  //    - indicate custom login is NOT in progress
   contents.on('did-start-navigation', (event, url) => {
     const contentID = event.sender.id;
-    const parsedUrl = new URL(url);
-    const urlIsTrusted = isTrustedURL(parsedUrl);
-    const urlIsCustomLoginPath = isCustomLoginURL(parsedUrl);
+    const parsedURL = parseURL(url);
+    const urlIsTrusted = isTrustedURL(parsedURL);
+    const urlIsCustomLoginPath = isCustomLoginURL(parsedURL);
     const previousPage = event.sender.history[event.sender.history.length - 1];
 
     if (urlIsTrusted && urlIsCustomLoginPath && !customLogins[contentID].inProgress && previousPage.endsWith('/login')) {
```

Comment:
Will this `endsWith` check fail if there are query string parameters? Is it strictly necessary to check `previousPage` at all?

Reply:
Good question. @crspeller, do you think it's necessary to ensure a SAML/OAUTH endpoint is accessed directly from the /login page in order to trigger the updated custom login process?

Reply:
Probably not necessary. This can be removed. Worried about some kind of deep login link failing because of this.

Reply:
K, sounds good, thanks!